app/lib/kubernetes_client.py

```diff
@@ -5,6 +5,104 @@
 namespace = get_namespace()
 
 
+def get_pod(pod_name):
+    pod = kubernetes_client.CoreV1Api().read_namespaced_pod(namespace=namespace, name=pod_name)
+    return pod
+
+
+def get_pod_details(pod_name):
+    pod = get_pod(pod_name)
+    volume_details = []
+
+    # Get the specs of each volume
+    for volume in pod.spec.volumes:
+        if volume.persistent_volume_claim:
+            pvc_name = volume.persistent_volume_claim.claim_name
+            volume_specs = get_pod_pvc_volume_size(namespace, pvc_name)
+            volume_details.append({
+                'pvc_name': pvc_name,
+                'size': volume_specs['size'],
+                'class': volume_specs['class'],
+                'phase': volume_specs['phase']
+            })
+
+
+    affinity = pod.spec.affinity
+    if affinity:
+        affinity_details = get_affinity_details(affinity)
+    else:
+        affinity_details = {}
+
+    # Get the node name where the pod is scheduled
+    pod_node_location = pod.spec.node_name
+    node_details = get_pod_node_details(pod_node_location)
+
+    potential_ready = pod.status.phase == "Running"
+    container_resources = pod.spec.containers[0].resources
+
+    pod_details = {
+        'affinity': affinity_details,
+        'limit_cpu': container_resources.limits.get('cpu', '') if container_resources.limits else 'None',
+        'limit_memory': container_resources.limits.get('memory', '') if container_resources.limits else 'None',
+        'name': pod.metadata.name,
+        'namespace': pod.metadata.namespace,
+        'node_details': node_details,
+        'node_name': pod.spec.node_name,
+        'node_selector': pod.spec.node_selector or {},
+        'potential_ready': potential_ready,
+        'request_cpu': container_resources.requests.get('cpu', '') if container_resources.requests else 'None',
+        'request_memory': container_resources.requests.get('memory', '') if container_resources.requests else 'None',
+        'restart_count': pod.status.container_statuses[0].restart_count,
+        'tolerations': pod.spec.tolerations or [],
+        'volume_details': volume_details
+    }
+    return pod_details
+
+
+def get_pod_pvc_volume_size(namespace, pvc_name):
+    pvc = kubernetes_client.CoreV1Api().read_namespaced_persistent_volume_claim(namespace=namespace, name=pvc_name)
+    pvc_details = {
+        'size': pvc.spec.resources.requests['storage'],
+        'class': pvc.spec.storage_class_name,
+        'phase': pvc.status.phase
+    }
+    return pvc_details
+
+
+def get_pod_node_details(pod_name):
+    node = kubernetes_client.CoreV1Api().read_node(name=pod_name)
+    node_details = {}
+
+    for label in ["beta.kubernetes.io/arch", "beta.kubernetes.io/instance-type", "beta.kubernetes.io/os"]:
```

Comment:
Maybe also add the non-beta labels to this list as it those are apparently deprecated already: https://kubernetes.io/docs/reference/labels-annotations-taints/#kubernetes-io-arch

Reply:
Nice! I removed the deprecated values and replaced with the new one